src/extensions/extra/selfie-taker/camera.js

```diff
@@ -0,0 +1,201 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true,  regexp: true, indent: 4, maxerr: 50 */
+/*global define, brackets, $, window */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var Dialog = brackets.getModule("widgets/Dialogs");
+    var Buffer = brackets.getModule("filesystem/impls/filer/BracketsFiler").Buffer;
+    var Strings = brackets.getModule("strings");
+    var fs = brackets.getModule("fileSystemImpl");
+    var selfieDialogHTML = require("text!dialog.html");
+    var selfieWidgetHTML = require("text!selfieWidget.html");
+    var _selfieDialog;
+    var video;
+    var photo;
+    var snap;
+    var snapAudio;
+    var canvas;
+    var use;
+    var deferred;
+    var filePath;
+    var streaming = false;
+    var height;
```

Comment:
I feel like you have way too many globals here.  I'll try to find places below where they aren't needed, but be aware of code like this.